### ISSUE TITLE
processes a ZIP of images, segments objects, separates them into "normal" and "abnormal" categories, and returns both types in a single output ZIP with organized folders.

### ISSUE THREAD

**Issue by fire** [V-Sekai-fire/godot-visual-anomaly-detection#2]
Rough draft.

```python
# LLM GENERATED
# NOT PROVEN YET
# Define variables for paths
INPUT_ZIP := "input.zip"
OUTPUT_ZIP := "output_images.zip"
EXTRACTED_FOLDER := "input_images"

# Define the command to unzip the input file
unzip_input:
    @echo "Unzipping the input ZIP file..."
    unzip $(INPUT_ZIP) -d $(EXTRACTED_FOLDER)

# Define the command to process images using SAM and EfficientAD
process_images:
    @echo "Processing images..."
    python3 -c "
import os
import torch
from segment_anything import SamPredictor, sam_model_registry
from anomalib.models.efficientad import EfficientAD
from PIL import Image
import shutil

device = 'cuda' if torch.cuda.is_available() else 'cpu'

# Load models
sam_model = sam_model_registry['vit_b'](checkpoint='path_to_sam_checkpoint.pth').to(device)
sam_predictor = SamPredictor(sam_model)
efficientad_model = EfficientAD('efficientnet_b0')

normal_images = []
abnormal_images = []

# Function to process each image
def process_image(file_path):
    image = Image.open(file_path).convert('RGB')
    sam_predictor.set_image(image)
    masks = sam_predictor.predict()

    for i, mask in enumerate(masks):
        object_img = image.crop(mask.bbox)
        obj_tensor = torch.from_numpy(object_img).unsqueeze(0).to(device)

        # Anomaly detection using EfficientAD
        with torch.no_grad():
            anomaly_score = efficientad_model(obj_tensor).item()
        
        # Classify based on anomaly score
        if anomaly_score < 0.5:
            normal_images.append((file_path, object_img))
        else:
            abnormal_images.append((file_path, object_img))

# Process images
input_folder = '$(EXTRACTED_FOLDER)'
for filename in os.listdir(input_folder):
    file_path = os.path.join(input_folder, filename)
    process_image(file_path)

# Save the output ZIP
from zipfile import ZipFile
with ZipFile('$(OUTPUT_ZIP)', 'w') as zip_file:
    for idx, (filename, image) in enumerate(normal_images):
        buffer = BytesIO()
        image.save(buffer, format='PNG')
        zip_file.writestr(f'normal/{filename}_{idx}.png', buffer.getvalue())
    
    for idx, (filename, image) in enumerate(abnormal_images):
        buffer = BytesIO()
        image.save(buffer, format='PNG')
        zip_file.writestr(f'abnormal/{filename}_{idx}.png', buffer.getvalue())

# Clean up
shutil.rmtree(input_folder)
    "

# Define the command to clean up
clean:
    @echo "Cleaning up temporary files..."
    rm -rf $(EXTRACTED_FOLDER)

# Create the output ZIP
zip_output:
    @echo "Creating output ZIP..."
    zip -r $(OUTPUT_ZIP) $(EXTRACTED_FOLDER)
``
